### PR TITLE
JHy coverage

### PR DESCRIPTION
**Pull request by x3mSpeedy**
**Add coverage functionality and more flags variables**

Add additional flags:
- Add flags for external libs
- Add PETSc and BOOST flags which inherits values from external libs and global (cmake) flags.
- Add macro simplifying setting variable value using other variables as possible options.
- Alter external library templates, now using finer flags specified above

Add new build type coverage for code coverage testing:
- Using coverage build type will compile flow123d with flags resulting in instrumented binary suitable for code coverage testing.
- Add cmake function for collection code coverage (xml or html format)
- Add yaml configuration for coverage build type
- Add python script for merging and filtering coverage (cobertura) reports

### CODE REVIEW

File: CMake/Modules/InstallPetsc.cmake [hunk 1]
```diff
@@ -104,7 +110,7 @@ file (WRITE "${cmakelists_fname}"
     URL ${PETSC_INSTALL_URL}
     SOURCE_DIR ${PETSC_INSTALL_DIR}/src
     BINARY_DIR ${PETSC_INSTALL_DIR}/src
-    CONFIGURE_COMMAND bash ${PETSC_INSTALL_DIR}/conf.sh --with-make-np ${MAKE_NUMCPUS}
+    CONFIGURE_COMMAND bash ${PETSC_INSTALL_DIR}/conf.sh
```
Comment (jbrezmorf) [inline CMake/Modules/InstallPetsc.cmake]: It should be correctly merged in order to keep changes in commit 22bd060.

File: unit_tests/jenkins_jobs_template.yaml [hunk 2]
```diff
@@ -414,5 +414,335 @@
...
+      - "Flow123d-{platform}-coveragetest-unit-test-{unit-dir}"
+      - "Flow123d-{platform}-coveragetest-coverage-unit-test"
+      - "Flow123d-{platform}-coveragetest-test-{test-dir}"
+      - "Flow123d-{platform}-coveragetest-coverage-test"
+      - "Flow123d-{platform}-coveragetest-multijob"
+
+
+####################################################################
+# Flow123d configure (coveragetest)
+- job-template:
```
Comment (jbrezmorf) [inline unit_tests/jenkins_jobs_template.yaml]: No need to introduce new job template, this fits to the general one.

**Comment by jbrezmorf** [comment]
Pretty good. 
Check corract merge with other commit.

**Comment by x3mSpeedy** [comment]
I cherry-picked your commit (568c6d4cb5780d796565034c15bfa8bd5ee4e7df) and only one file was affected.

**Pull request merged**